sonic_y_cable/y_cable.py

```diff
@@ -1334,24 +1337,67 @@ def get_nic_voltage(physical_port):
     return voltage
 
 
-def upgrade_firmware(physical_port, fwfile):
-    """ This routine should facilitate complete firmware
-    upgrade of the Y cable on all the three ends of the
-    Y cable of the port specified.
-    All the components of the Y cable should be upgraded and committed
-    in their entirety by this single call subroutine.
-    This should return success code if firmware upgrade is successful
-    and an error code otherwise.
+def download_firmware(physical_port, fwfile):
+    """ This routine should download and store the firmware on all the
+    components of the Y cable of the port specified.
+    This should include any internal transfers, checksum validation etc.
+    from TOR to TOR or TOR to NIC side of the firmware specified by the fwfile.
+    This basically means that the firmware which is being downloaded should be
+    available to be activated (start being utilized by the cable) once this API is
+    successfully executed.
+    Note that this API should ideally not require any rollback even if it fails
+    as this should not interfere with the existing cable functionality because
+    this has not been activated yet.
 
     Args:
         physical_port:
              an Integer, the actual physical port connected to a Y cable
         fwfile:
-             a string, a path to the binary file which contains the firmware image
+             a string, a path to the binary file which contains the firmware image.
```

Comment:
Remove 'binary' from this line now that we have a detailed description of the file format.

Reply:
fixed